CorsixTH/Lua/entities/humanoids/vip.lua

```diff
@@ -126,6 +180,55 @@ function Vip:goHome()
   self:despawn()
 end
 
+function Vip:evaluateRoom()
+  local room_extinguisher = 0
+  local room_plant = 0
+  local room_bin = 0
+  -- Another room visited.
+  self.num_visited_rooms = self.num_visited_rooms + 1
+  local room = self.next_room
+  -- if the player is about to kill a live patient for research, punish hard
+  if room.room_info.id == "research" then
+    if room:getPatient() then
+      self.vip_rating = self.vip_rating + 6
+    end
+  end
+  -- evaluate the room we're currently looking at
+  for object, _ in pairs(room.objects) do
+    if object.object_type.id == "extinguisher" and room_extinguisher == 0 then
+      self.room_eval = self.room_eval + 1
+      -- only count this object type once
+      room_extinguisher = 1
+    elseif object.object_type.id == "plant" then
```

Comment:
This is the state of the plant.  After talking to @Alberth289346  to make it easier to read and to test consider adding plant:isDying() function or similar.

Reply:
We have Plant:needsWatering()

```lua
function Plant:needsWatering()
  if self.current_state == 0 then
    if self.days_left < 10 then
      return true
    end
  else
    return true
  end
end
```

Reply:
Though I think there's an implication the player is unfairly punished because a plan just called for watering and handyman hasn't been given a chance to respond yet